### DISCUSSION TITLE
HttpEndPoint activity conditional bad request

### DISCUSSION THREAD

**Discussion by ricksearcy** [elsa-workflows/elsa-core#1260] [Q&A]
Using the Approval workflow as an example where I create 2 branches for Approve, Reject and each creates an HttpEndPoint activity. I have a use case where under certain conditions I need to return a 400 bad request when someone calls the Approve or Reject endpoints, but I don't want the workflow to move forward or the endpoints to go out of scope. Basically a 400 response, but the workflow remains in the same state as before the request was made. I've tried numerous methods including your example "BrickWallActivity" and "GoBackResult", but nothing I have tried allows the endpoints to remain. Can you point me in the right direction?

**Comment by sfmskywalker** [comment]
Sure, the easiest way is to simply reconnect from the HTTP Response activity back to the HTTP Endpoint activity.

Will that work for your scenario? If not, I’d be curious to hear about your use case.

**Reply by ricksearcy** [reply]
"Sure, the easiest way is to simply reconnect from the HTTP Response activity back to the HTTP Endpoint activity."

Could you show an example of that?

Use case: 
1. Approve and Reject endpoints are created for an approval workflow. There is a requestor and a list of approvers. 
2. The only users that should be allowed to make an approve or reject request are in the list of approvers. If anyone else attempts to do so I would like return a bad request, but not end the workflow.

It really could be any case where the  http request does not meet some defined business rule. Where a response is sent, but the workflow is uninterrupted.

**Reply by sfmskywalker** [reply]
Like this? This workflow connects back to the HTTP Endpoint activity after writing a Bad Request response:

![image](https://user-images.githubusercontent.com/938393/127029118-787e991d-894c-485c-83e3-504de562e410.png)

**Reply by ricksearcy** [reply]
Yes, that appears to be what I was trying to accomplish, but all in code.

**Reply by sfmskywalker** [reply]
I see. In that case, you can connect to arbitrary activities by using the `ThenNamed` method.
For example:

```csharp
builder
.HttpEndpoint().WithName("ApproveEndpoint")
.WriteHttpResponse("Bad human")
.ThenNamed("ApproveEndpoint")
```
